src/validation.cpp

```diff
@@ -4393,6 +4424,12 @@ static bool ContextualCheckBlockHeader(const CBlockHeader& block, CValidationSta
             return state.Invalid(false, REJECT_OBSOLETE, strprintf("bad-version(0x%08x)", block.nVersion),
                                  strprintf("rejected nVersion=0x%08x block", block.nVersion));
 
+    if (!block.fProofOfStake && block.nTime >= Params().PowUpdateTimestamp()) {
+        if (!block.IsSha256D() && !block.IsRandomX() && !block.IsProgPow())
+            return state.DoS(33, error("%s - bad-pow-algo-use-updated-algos", __func__), REJECT_OBSOLETE,
+                                 "must use progpow, randomx, or sha256d to mine blocks now");
```

Comment:
Be specific about "now", i.e "since block x hard fork"

Reply:
okay. sure. it is a timestamp. but once we have the block height we can update this. What do you think?

Reply:
I'm not sure it's necessary to put a block number, or date in there.  If they are hitting that, they're trying to use pre-fork mining algos against the post fork chain.  Since we can enforce the fork at the protocol level anyway and not rely on this DoS protection check, they don't really need it to be pretty and nice.  They're doing something they shouldn't be doing, and if they actually hit that message... they either -know- they're trying something nefarious, or .. we've utterly failed marketing.

Reply:
Ok, so "unexpected hash, please ensure its progpow, randomx, or sha256d." or something?

Reply:
Done